src/tensor_ops/map.rs

```diff
@@ -272,6 +314,32 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_leakyrelu() {
+        let x = tensor([-2.0, -1.0, 0.0, 1.0, 2.0]);
+        let r = x.trace().leakyrelu();
+        assert_eq!(r.data(), &[-0.02, -0.01, 0.0, 1.0, 2.0]);
+        // NOTE: call .exp() to make sure we cover cases where .leakyrelu() uses the result's gradient
+        let gradients = backward(r.exp().mean());
+        assert_eq!(
+            gradients.ref_gradient(&x),
+            &[0.0019603972, 0.0019800996, 0.002, 0.54365635, 1.4778112]
+        );
+    }
+
+    #[test]
+    fn test_softplus() {
+        let x = tensor([-2.0, -1.0, 0.0, 1.0, 2.0]);
+        let r = x.trace().softplus();
+        assert_eq!(r.data(), &[0.12692805, 0.31326166, 0.6931472, 1.3132616, 2.126928]);
+        // NOTE: call .exp() to make sure we cover cases where .softplus() uses the result's gradient
+        let gradients = backward(r.exp().mean());
+        assert_eq!(
+            gradients.ref_gradient(&x),
+            &[0.12072917, 0.15803963, 0.26666668, 0.58604467, 1.499113]
```

Comment:
Ran this through pytorch and should be close to:
```suggestion
            &[0.02706706, 0.07357589, 0.20000000, 0.54365635, 1.47781146]
```